BUG Interactive authentication not supported on Mac OS

**Describe the bug**
Interactive authentication currently does not support mac os. It is unable to build on mac os due to tao::platform::unix not being available.

The target_family is set to `unix` which is included in target arch `aarch64-apple-darwin`. https://github.com/sreeise/graph-rs-sdk/blob/7aa756af45fafec43156d798fff1c47c3fb04b9b/graph-oauth/src/interactive/interactive_auth.rs#L13-L14

Tao however only allows the following:
https://github.com/tauri-apps/tao/blob/0af78a924958f0b91f94fe58c78f26569cffdf5f/src/platform/unix.rs#L5-L11

**To Reproduce**
Steps to reproduce the behavior:
1. Compile on mac os.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**

```
error[E0432]: unresolved import `tao::platform::unix`
  --> /src/index.crates.io-6f17d22bba15001f/graph-oauth-2.0.0/src/interactive/interactive_auth.rs:14:20
   |
14 | use tao::platform::unix::EventLoopBuilderExtUnix;
   |                    ^^^^ could not find `unix` in `platform`
```

**Desktop (please complete the following information):**
 - OS: Mac OS M1

--- a/graph-oauth/src/interactive/interactive_auth.rs
+++ b/graph-oauth/src/interactive/interactive_auth.rs
@@ -10,8 +10,12 @@ use tao::window::{Window, WindowBuilder};
 use url::Url;
 use wry::WebView;
 
+#[cfg(not(target_os = "macos"))]
 #[cfg(target_family = "unix")]
 use tao::platform::unix::EventLoopBuilderExtUnix;
+
+#[cfg(target_os = "macos")]
+use tao::platform::macos::{EventLoopWindowTargetExtMacOS, WindowBuilderExtMacOS, WindowExtMacOS};
 
 #[cfg(target_family = "windows")]
 use tao::platform::windows::EventLoopBuilderExtWindows;
@@ -180,6 +184,7 @@ where
             .with_theme(options.theme)
     }
 
+    #[cfg(not(target_os = "macos"))]
     #[cfg(target_family = "unix")]
     fn window_builder(options: &WebViewOptions) -> WindowBuilder {
         WindowBuilder::new()
@@ -192,9 +197,28 @@ where
             .with_resizable(true)
     }
 
+    #[cfg(target_family = "macos")]
+    fn window_builder(options: &WebViewOptions) -> WindowBuilder {
+        WindowBuilder::new()
+            .with_title(options.window_title.clone())
+            .with_closable(true)
+            .with_content_protection(true)
+            .with_minimizable(true)
+            .with_maximizable(true)
+            .with_focused(true)
+            .with_resizable(true)
+    }
+
+    #[cfg(any(target_family = "windows", target_family = "unix"))]
+    #[cfg(not(target_os = "macos"))]
     fn event_loop() -> EventLoop<UserEvents> {
         EventLoopBuilder::with_user_event()
             .with_any_thread(true)
             .build()
+    }
+
+    #[cfg(target_os = "macos")]
+    fn event_loop() -> EventLoop<UserEvents> {
+        EventLoopBuilder::with_user_event().build()
     }
 }

